### ISSUE TITLE
SDKMAN should to be installable without profile modifications

### ISSUE THREAD

**Issue by andvgal** [sdkman/sdkman-cli#556]
There are many cases including CI servers which run all sorts of package version managers and it may not be welcome to have SDKMAN/nvm/rvm/{name other} be loaded into environment by default.

Please wrap the following sections of the installer:
```bash
echo "Attempt update of bash profiles..."
echo "Attempt update of zsh profiles..."
```
with something like:
```bash
if [ -z "$SDKMAN_SKIP_PROFILE" ]; then
    # ...
fi
```

**Comment by devinsba** [comment]
Agreed, I had added something like this to my dotfiles so I can install it and not get my zshrc modified but apparently the string I was searching for doesn't work anymore..

**Comment by marc0der** [comment]
Given that the installer is a dynamic bash script that is served from the [sdkman-hooks](https://github.com/sdkman/sdkman-hooks) app, it might make sense to introduce a request parameter that allows for this. Something like:

    curl -s https://get.sdkman.io?noprofile | bash

Would this work?

**Comment by marc0der** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Install without profile modifications](https://trello.com/c/1vC5uCRS/29-install-without-profile-modifications)

**Comment by Bessonov** [comment]
I use sdkman to install packages inside docker images. The problem is, that entries are added to `.bashrc` and therefore doesn't work in non-interactive environment. As workaround I have

```
RUN echo ". $HOME/.sdkman/bin/sdkman-init.sh" >> ~/.profile
```

Therefore I think it's better to not disable profile modification, but provide a greater control of it. For example like:
```
# doesn't modify profiles
curl -s https://get.sdkman.io?profile=none | bash

# modify first existing file
curl -s https://get.sdkman.io?profile=.profile,.bashrc | bash

# modify all existing files
curl -s https://get.sdkman.io?profile=.profile,.bashrc&strategy=all | bash

# modify or create a file
curl -s https://get.sdkman.io?profile=.bash_aliases&create=true | bash
```

Maybe allow paths like `~/.profile` or `/etc/bash.bashrc`.

@devinsba Thanks for sharing your workaround.

**Comment by marc0der** [comment]
The request parameter to control profile updates sound like a good idea, and would need to be implemented on the [sdkman-hooks](https://github.com/sdkman/sdkman-hooks) project.

**Comment by EugenMayer** [comment]
Also needed here, in my case i manage SDKs using sdkman but manage the shell integration with jenv - i install sdkman / jdks using ansible and well modifing my ZSH is nothing i desire here. Thank you for working on that

**Comment by apottere** [comment]
It looks the sdkman-hooks issue for this work is https://github.com/sdkman/sdkman-hooks/issues/16
